frontend/public/components/utils/router-hooks.ts

```diff
@@ -13,6 +13,8 @@ export const usePreventDataLossLock = (lock: boolean) => {
     };
     if (lock) {
       window.addEventListener('beforeunload', onBeforeUnload);
+    } else {
+      window.removeEventListener('beforeunload', onBeforeUnload);
```

Comment:
Unnecessary because if `lock` changed, the effect dispose function would have already removed the listener. Furthermore the function `onBeforeUnload` is created within this closure and would never have been added as a window listener.